### PR TITLE
vlan: Add a function to get a free port

### PR DESCRIPTION
**Pull request by yanglei-rh**
1. Add a function to get a free port on the dst guest

ID:2078435

Signed-off-by: Lei Yang <leiayang@redhat.com>

### CODE REVIEW

**Comment by yanglei-rh** [comment]
Test Pass:
 (2/3) Host_RHEL.m8.u7.product_rhel.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vlan.q35: PASS (700.23 s)
 (3/3) Host_RHEL.m8.u7.product_rhel.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vlan.vlan_scalability_test.q35: PASS (2471.54 s)

**Comment by yanglei-rh** [comment]
Hello @PaulYuuu  Could you please help me review it? Thanks a lot.

File: generic/tests/vlan.py [hunk 1]
```diff
@@ -91,11 +91,22 @@ def rem_vlan(test, session, v_id, iface="eth0", cmd_type="ip"):
                               test.log.info)
         return session.cmd_status(rem_vlan_cmd)
 
+    def find_free_port():
+        """
+        Returns a free port on the host
+        """
+        check_cmd = "netstat -nultp |awk '(NR>2){print $4}' | awk -F':' '{print $NF}'"
+        get_used_port = process.run(check_cmd, verbose=False, shell=True).stdout.decode()
```
[the remark this answers is not in the document]
Comment (yanglei-rh) [reply generic/tests/vlan.py]: > In this case, it is to get the free port in the guest.

Thanks for your reminder,this line has been changed the guest session. Please help me review it again.

File: generic/tests/vlan.py [hunk 2]
```diff
@@ -91,11 +90,21 @@ def rem_vlan(test, session, v_id, iface="eth0", cmd_type="ip"):
                               test.log.info)
         return session.cmd_status(rem_vlan_cmd)
 
+    def find_free_port(session):
+        """
+        Returns a free port on the host
```
Comment (PaulYuuu) [inline generic/tests/vlan.py:95]: Doc typo, should "on the guest"

File: generic/tests/vlan.py [hunk 3]
```diff
@@ -91,11 +90,21 @@ def rem_vlan(test, session, v_id, iface="eth0", cmd_type="ip"):
...
+        """
+        check_cmd = "netstat -nultp |awk '(NR>2){print $4}' | awk -F':' '{print $NF}'"
+        used_port_names = session.cmd_output(check_cmd).strip().split('\n')
```
Comment (PaulYuuu) [inline generic/tests/vlan.py:98]:
```suggestion
        used_port_names = session.cmd_output(check_cmd).strip().splitlines()
```

File: generic/tests/vlan.py (hunk 2 above)
Comment (yanglei-rh) [reply generic/tests/vlan.py]: Changed, thanks.

File: generic/tests/vlan.py (hunk 3 above)
Comment (yanglei-rh) [reply generic/tests/vlan.py]: Please help review it again, this line has been replaced, thanks a lot.

File: generic/tests/vlan.py [hunk 4]
```diff
@@ -91,11 +90,21 @@ def rem_vlan(test, session, v_id, iface="eth0", cmd_type="ip"):
...
+        port_range = list(range(1, 65535))
+        free_ports = list(set(used_port_names).difference(set(port_range)))
+        return free_ports[0]
+
     def nc_transfer(test, src, dst):
         """
         Transfer file by netcat
         """
-        nc_port = utils_misc.find_free_port(1025, 5334, vm_ip[dst])
+        nc_port = find_free_port(session)
```
Comment (PaulYuuu) [inline generic/tests/vlan.py:107]: Where is the free port you want to find? In src? Or dst? `session` is not unique.

Comment (yanglei-rh) [reply generic/tests/vlan.py]: It should be the session of src, I have made changes to the code, please help to review it again.

**Comment by yanglei-rh** [comment]
Test Pass
(1/2) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vlan.q35: PASS (693.09 s)
 (2/2) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vlan.vlan_scalability_test.q35: PASS (2467.45 s)

File: generic/tests/vlan.py [hunk 5]
```diff
@@ -91,11 +90,21 @@ def rem_vlan(test, session, v_id, iface="eth0", cmd_type="ip"):
...
+        port_range = list(range(1025, 65535))
+        free_ports = list(set(port_range).difference(set(used_port_names)))
+        return free_ports[0]
+
     def nc_transfer(test, src, dst):
         """
         Transfer file by netcat
         """
-        nc_port = utils_misc.find_free_port(1025, 5334, vm_ip[dst])
+        nc_port = find_free_port(src)
```
Comment (PaulYuuu) [inline generic/tests/vlan.py:107]: You can see `utils_misc.find_free_port(1025, 5334, vm_ip[dst])`, **dst**, so the nc_port is a free port in the dst guest.

Comment (yanglei-rh) [reply generic/tests/vlan.py]: Thanks for your reminder, I have always misunderstood it as the ip address in the nc command. `src` has been replaced to `dst`, please help review it again, thanks a lot.

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**